operator/webhook/core/test/test_sync.py

```diff
@@ -189,6 +235,138 @@ def test_env_vars_no_keystore(full_route):
+        "pkcs12": {
+            "secretName": "test-tls-secret",
+            "key": "test-keystore.p12",
+            "passwordSecretRef": "keystore-password-ref",
+        }
+    }
+    expected_keystore_volume = {
+        "name": "keystore",
+        "secret": {
+            "secretName": "test-tls-secret",
+            "items": [{"key": "test-keystore.p12", "path": "test-keystore.p12"}],
+        },
+    }
+
+    expected_truststore_volume = {
+        "name": "truststore",
+        "configMap": {
+            "name": "test-tls-cm",
+            "items": [{"key": "test-truststore.p12", "path": "test-truststore.p12"}],
+        },
+    }
+
+    response = sync(full_route)
+    volumes = response["children"][0]["spec"]["template"]["spec"]["volumes"]
+    actual_keystore_volume = None
+    actual_truststore_volume = None
+    for volume in volumes:
+        if volume["name"] == "keystore":
+            actual_keystore_volume = volume
+        elif volume["name"] == "truststore":
+            actual_truststore_volume = volume
+
+    assert actual_keystore_volume == expected_keystore_volume
+    assert actual_truststore_volume == expected_truststore_volume
+
+
+def test_volume_jks_keystore_and_jks_truststore(full_route):
+    del full_route["parent"]["spec"]["tls"]["truststore"]
+    full_route["parent"]["spec"]["tls"]["truststore"] = {
+        "configMapName": "test-tls-cm",
+        "key": "test-truststore.jks",
+        "type": "jks",
+    }
+    expected_keystore_volume = {
+        "name": "keystore",
+        "secret": {
+            "secretName": "test-tls-secret",
+            "items": [{"key": "test-keystore.jks", "path": "test-keystore.jks"}],
+        },
+    }
+
+    expected_truststore_volume = {
+        "name": "truststore",
+        "configMap": {
+            "name": "test-tls-cm",
+            "items": [{"key": "test-truststore.jks", "path": "test-truststore.jks"}],
+        },
+    }
+    response = sync(full_route)
+    volumes = response["children"][0]["spec"]["template"]["spec"]["volumes"]
+    actual_keystore_volume = None
+    actual_truststore_volume = None
+    for volume in volumes:
+        if volume["name"] == "keystore":
+            actual_keystore_volume = volume
+        elif volume["name"] == "truststore":
+            actual_truststore_volume = volume
+
+    assert actual_keystore_volume == expected_keystore_volume
+    assert actual_truststore_volume == expected_truststore_volume
+
+
+def test_jks_keystore_custom_key_alias(full_route):
+    full_route["parent"]["spec"]["tls"]["keystore"]["jks"]["alias"] = "mykeyalias"
+    expected_ssl_config_json = json.dumps(
+        {
+            "ssl": {
+                "key-alias": "mykeyalias",
+                "key-store": "/etc/keystore/test-keystore.jks",
+                "key-store-type": "JKS",
+            },
+            "port": 8443,
+        }
+    )
+    actual_ssl_config_json = json.dumps(_get_server_ssl_config(full_route["parent"]))
```

Comment:
:question: We do we convert the dictionaries to `json` before comparing?

Reply:
Removed json conversion